internal/provider/auth_gcp.go

```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/hashicorp/terraform-provider-vault/internal/consts"
 )
 
+const defaultGCPScope = "https://www.googleapis.com/auth/cloud-platform"
```

Comment:
I wonder if it would make sense to add a new scheme field for configuring the scopes in the `provider` resource block. It could default to `[]string{defaultGCPScope}`?

Reply:
I am not sure I understand the need behind configurable scopes. Since Vault requires the default scope to be set, what value would another scope bring here ? 
In other words, if the token was created with an additional scope, would Vault/the provider use this scope ?

You know this part way better than I do, so I'll follow your advice, but I'd like to understand :)

Reply:
Good point. I think we can keep it in sync with the way that `vault login method=gcp` works today. If we ever need to support configuring "other" scopes then we have pretty easy path forward.